pkg/handler/iam_handler_test.go

```diff
@@ -1102,6 +1102,507 @@ func TestDo(t *testing.T) {
+								Condition: &expr.Expr{
+									Title:      defaultConditionTitle,
+									Expression: fmt.Sprintf("request.time <= timestamp('%s')", now.Add(-1*time.Hour).Format(time.RFC3339)),
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOrganizationsPolicy: &iampb.Policy{
+				Bindings: []*iampb.Binding{
+					{
+						Members: []string{
+							"user:test-org-userB@example.com",
+						},
+						Role: "roles/accessapproval.approver",
+						Condition: &expr.Expr{
+							Title:      defaultConditionTitle,
+							Expression: fmt.Sprintf("request.time <= timestamp('%s')", now.Add(-1*time.Hour).Format(time.RFC3339)),
+						},
+					},
+				},
+			},
+			wantFoldersPolicy:  &iampb.Policy{},
+			wantProjectsPolicy: &iampb.Policy{},
+			wantErrSubstr:      "failed to check expiry",
+		},
+		{
+			name: "failed_clean_up_expired_bindings_wrong_expiration",
+			organizationsServer: &fakeServer{
+				policy: &iampb.Policy{
+					Bindings: []*iampb.Binding{
+						{
+							Members: []string{
+								"user:test-org-userB@example.com",
+							},
+							Role: "roles/accessapproval.approver",
+							Condition: &expr.Expr{
+								Title:      defaultConditionTitle,
+								Expression: fmt.Sprintf("request.time < timestamp('%s')", now.Add(-1*time.Hour).Format(time.RFC850)),
+							},
+						},
+					},
+				},
+			},
+			foldersServer: &fakeServer{
+				policy: &iampb.Policy{},
+			},
+			projectsServer: &fakeServer{
+				policy: &iampb.Policy{},
+			},
+			request: &v1alpha1.IAMRequest{
+				ResourcePolicies: []*v1alpha1.ResourcePolicy{
+					{
+						Resource: "organizations/foo",
+						Bindings: []*v1alpha1.Binding{
+							{
+								Members: []string{
+									"user:test-org-userA@example.com",
+								},
+								Role: "roles/bigquery.dataViewer",
+							},
+						},
+					},
+				},
+			},
+			wantPolicies: []*v1alpha1.IAMResponse{
+				{
+					Resource: "organizations/foo",
+					Policy: &iampb.Policy{
+						Bindings: []*iampb.Binding{
+							{
+								Members: []string{
+									"user:test-org-userB@example.com",
+								},
+								Role: "roles/accessapproval.approver",
+								Condition: &expr.Expr{
+									Title:      defaultConditionTitle,
+									Expression: fmt.Sprintf("request.time < timestamp('%s')", now.Add(-1*time.Hour).Format(time.RFC850)),
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOrganizationsPolicy: &iampb.Policy{
+				Bindings: []*iampb.Binding{
+					{
+						Members: []string{
+							"user:test-org-userB@example.com",
+						},
+						Role: "roles/accessapproval.approver",
+						Condition: &expr.Expr{
+							Title:      defaultConditionTitle,
+							Expression: fmt.Sprintf("request.time < timestamp('%s')", now.Add(-1*time.Hour).Format(time.RFC850)),
+						},
+					},
+				},
+			},
+			wantFoldersPolicy:  &iampb.Policy{},
+			wantProjectsPolicy: &iampb.Policy{},
+			wantErrSubstr:      "failed to check expiry",
+		},
+		{
+			name: "ignore_non-AOD_bindings",
```

Comment:
How about adding a case for ignoring non-requester bindings?

Reply:
Added